video2dataset/workers/optical_flow_worker.py

```diff
@@ -0,0 +1,255 @@
+"""
+Worker for optical flow stage
+"""
+import time
+import pyarrow as pa
+import traceback
+import io
+import numpy as np
+import fsspec
+import tempfile
+import os
+import cv2
+
+from video2dataset.logger import CappedCounter, write_stats
+from video2dataset.subsamplers import OpticalFlowSubsampler
+from video2dataset.dataloader import get_video_dataset
+
+
+def numpy_npz_dumps(numpy_dict):
+    """
+    Dump a dictionary of numpy arrays into a bytestring using numpy npz format.
+
+    Args:
+        numpy_dict (dict): A dictionary containing numpy arrays as values.
+
+    Returns:
+        bytes: A bytestring representing the compressed numpy arrays.
+    """
+
+    stream = io.BytesIO()
+    np.savez_compressed(stream, **numpy_dict)
+    return stream.getvalue()
+
+
+def frames_to_mp4_bytes(frames, fps=30, codec="mp4v"):
+    """
+    Convert a list of frames to an mp4 video encoded as bytes.
+
+    Args:
+        frames (list): A list of frames.
+        fps (int): The frames per second of the video. Default is 30.
+        codec (str): The codec to use for encoding. Default is 'mp4v'.
+
+    Returns:
+        bytes: A bytestring representing the encoded video.
+    """
+    # Get frame dimensions from the first frame
+    height, width, _ = frames[0].shape
+
+    # Create a temporary file to save the video
+    with tempfile.NamedTemporaryFile(suffix=".mp4", delete=False) as tmpfile:
+        video_path = tmpfile.name
+
+        # Define the codec and create a VideoWriter object
+        fourcc = cv2.VideoWriter_fourcc(*codec)
+        out = cv2.VideoWriter(video_path, fourcc, fps, (width, height))
+
+        # Write the frames to the video
+        for frame in frames:
+            out.write(frame)
+
+        # Release the VideoWriter
+        out.release()
+
+        # Read the video file as bytes
+        with open(video_path, "rb") as f:
+            video_bytes = f.read()
+
+    # Remove the temporary video file
+    os.remove(video_path)
+
+    return video_bytes
+
+
+def convert_frames_depth(frames, target_depth=np.uint8):
```

Comment:
same why do we need this?

Reply:
converts stuff to uint8 so that it can be written as an mp4 bytestream